src/main/java/baseball/Application.java

```diff
@@ -1,7 +1,210 @@
+     */
+    public static List<Integer> evaluateGuess (List<Integer> guess, List<Integer> answer) {
+        int ballCount = countBall(guess, answer);
+        int strikeCount = countStrike(guess, answer);
+        return Arrays.asList(ballCount, strikeCount);
+    }
+
+    public static int countBall (List<Integer> guess, List<Integer> answer) {
+        int countOfCommonNumbers = countCommonNumber(guess, answer);
+        int countOfNumbersInSamePosition = countNumbersInSamePosition(guess, answer);
+        return countOfCommonNumbers - countOfNumbersInSamePosition;
+    }
+
+    public static int countStrike (List<Integer> guess, List<Integer> answer) {
+        return countNumbersInSamePosition(guess, answer);
+    }
+
+    public static int countCommonNumber (List<Integer> guess, List<Integer> answer) {
+        return (int) guess.stream().filter(answer::contains).count();
+
+    }
+
+    public static int countNumbersInSamePosition (List<Integer> guess, List<Integer> answer) {
+        return (int) IntStream.range(0, LENGTH_OF_NUMBERS)
+                .filter(i -> Objects.equals(guess.get(i), answer.get(i)))
+                .count();
+    }
+
+    public static List<Integer> stringToNumberList (String numbersInString) {
+        List<Integer> numberList = new ArrayList<>();
+        int number;
+        for ( int i = 0; i < LENGTH_OF_NUMBERS; i++ ) {
+            number = numbersInString.charAt(i) - '0';
+            numberList.add(number);
+        }
+        return numberList;
+    }
+
+    public static boolean isValidNumbersForGame (String input) {
+        if (!isOnlyNumbers(input)) {
+            return false;
+        }
+
+        if (!hasCorrectSize(input)) {
+            return false;
+        }
+
+        if (!isAllUnique(input)) {
+            return false;
+        }
+
+        if (hasZero(input)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static boolean isOnlyNumbers (String input) {
+        return Pattern.matches("^[0-9]*$", input);
+    }
+
+    public static boolean hasCorrectSize (String input) {
+        return input.length() == LENGTH_OF_NUMBERS;
+    }
+
+    public static boolean isAllUnique (String input) {
+        Set<Character> uniqueNumbers = new HashSet<>();
+        for (Character number : input.toCharArray()) {
+            uniqueNumbers.add(number);
+        }
+        return uniqueNumbers.size() == LENGTH_OF_NUMBERS;
+    }
+
+    public static boolean hasZero (String input) {
+        return Pattern.matches(".*0.*", input);
+    }
+
+    public static List<Integer> makeAnswer() {
+        List<Integer> answer = new ArrayList<>();
+        int randomNumber;
+        while ( answer.size() < LENGTH_OF_NUMBERS ) {
+            randomNumber = Randoms.pickNumberInRange(MIN_DIGIT, MAX_DIGIT);
+            if (answer.contains(randomNumber)) {
+                continue;
+            }
+            answer.add(randomNumber);
+        }
```

Comment:
```suggestion
        while ( answer.size() < LENGTH_OF_NUMBERS ) {
            int randomNumber = Randoms.pickNumberInRange(MIN_DIGIT, MAX_DIGIT);
            if (answer.contains(randomNumber)) {
                continue;
            }
            answer.add(randomNumber);
        }
```

변수를 사용할때 반복문 안에서 선언하는건 어떨까요?

참고 : https://dazbee.tistory.com/52